addons/web/static/tests/legacy/views/kanban_tests.js

```diff
@@ -2794,6 +2794,43 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('Open record when clicking on widget field', async function (assert) {
+        assert.expect(2);
+
+        var kanban = await createView({
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: '<kanban class="o_kanban_test">' +
+                        '<templates>' +
+                            '<t t-name="kanban-box">' +
+                                '<div class="oe_kanban_global_click">' +
+                                    '<field name="salary" widget="monetary"/>' +
+                                    '<div>' +
+                                        '<a class="o_test_link" href="#">test link</a>' +
+                                    '</div>' +
+                                '</div>' +
+                            '</t>' +
+                        '</templates>' +
+                    '</kanban>',
+            archs: {
+                'product,false,form': '<form string="Product"><field name="display_name"/></form>',
+            },
+            intercepts: {
+                switch_view: function (ev) {
+                    assert.deepEqual(_.pick(ev.data, 'res_id', 'view_type'), {
```

Comment:
I think it is equivalent to:
```suggestion
                    assert.deepEqual({
                        res_id: ev.data.res_id, 
                        view_type: ev.data.view_type,
                    }, {
```
or also like this:
```suggestion
                    const { res_id, view_type } = ev.data;
                    assert.deepEqual({ res_id, view_type }, {
```
🙂